src/Psl/Encoding/Base64/Internal/Base64.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Encoding\Base64\Internal;
+
+use Psl\Encoding\Exception;
+use Psl\Regex;
+use Psl\Str\Byte;
+
+use function pack;
+use function unpack;
+
+/**
+ * The following class was derived from code of Paragon Initiative Enterprises.
+ *
+ * https://github.com/paragonie/constant_time_encoding/blob/198317fa6db951dd791be0740915dae878f34b3a/src/Base64.php
+ *
+ * Code subject to MIT License (https://github.com/paragonie/constant_time_encoding/blob/198317fa6db951dd791be0740915dae878f34b3a/LICENSE.txt)
+ *
+ * Copyright (c) 2016 - 2022 Paragon Initiative Enterprises
+ */
+abstract class Base64
+{
+    /**
+     * Convert a binary string into a base64-encoded string.
+     *
+     * Base64 character set:
+     *  [A-Z]      [a-z]      [0-9]      +     /
+     *  0x41-0x5a, 0x61-0x7a, 0x30-0x39, 0x2b, 0x2f
+     *
+     * @pure
+     */
+    public static function encode(string $binary, bool $padding = true): string
+    {
+        $dest = '';
+        $binaryLength = Byte\length($binary);
```

Comment:
`Binary::safeStrlen` used in `paragonie/constant_time_encoding` refers to `mb_strlen($binary, '8bit')`, the equivalent in PSL would be `Str\length($binary, encoding: Str\Encoding::ASCII_8BIT);`, not `Byte\length($binary)`.

```suggestion
        $binaryLength = Str\length($binary, encoding: Str\Encoding::ASCII_8BIT);
```

( same applies to `Binary::safeSubstr` being `Str\slice($_, $_, $_, encoding: Str\Encoding::ASCII_8BIT)` )